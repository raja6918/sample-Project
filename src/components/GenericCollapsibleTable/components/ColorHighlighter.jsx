import React from 'react';
import PropTypes from 'prop-types';

const ColorHighlighter = ({ highlighter, value, data, style }) => {
  const customeStyle = highlighter(value, data);
  return <span style={customeStyle ? style : {}}>{value}</span>;
};

ColorHighlighter.propTypes = {
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  data: PropTypes.shape().isRequired,
  style: PropTypes.shape(),
  highlighter: PropTypes.func,
};

ColorHighlighter.defaultProps = {
  style: { color: '#FF650C' },
  highlighter: () => false,
};

export default ColorHighlighter;
