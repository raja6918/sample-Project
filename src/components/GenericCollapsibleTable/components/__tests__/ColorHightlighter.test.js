import React from 'react';
import { shallow } from 'enzyme';
import ColorHighlighter from '../ColorHighlighter';

const mockProps = {
  value: 'Test',
  data: { name: 'Test' },
  style: { color: '#FF650C' },
  highlighter: () => true,
};

test('ColorHighlighter Component render correctly', () => {
  const wrapper = shallow(<ColorHighlighter {...mockProps} />);
  expect(wrapper).toMatchSnapshot();
});
