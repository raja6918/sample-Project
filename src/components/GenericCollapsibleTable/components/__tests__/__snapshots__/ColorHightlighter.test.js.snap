// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorHighlighter Component render correctly 1`] = `
<span
  style={
    Object {
      "color": "#FF650C",
    }
  }
>
  Test
</span>
`;
